src/modules/roleSchemas.ts

```diff
@@ -30,6 +30,30 @@ export const removeRoleFromMembershipSchema = z.object({
 });
 export type RemoveRoleFromMembership = z.infer<typeof removeRoleFromMembershipSchema>;
 
+export const addScopeToRoleSchema = z.object({
+    code: z.string(),
+    scope: z.object({
+        // scopes by model schema
+        field: z.enum([
```

Comment:
we can do `z.string().refine((s) => scopes.includes(s))` to avoid copying all scopes